lib/charms/bind/v0/dns_record.py

```diff
@@ -245,15 +248,68 @@ class RequirerDomain(BaseModel):
     Attributes:
         domain: the domain name.
         username: username for authentication.
+        password: the password for authentication.
         password_id: secret password for authentication.
         uuid: UUID for this entry.
     """
 
     domain: str = Field(min_length=1)
     username: str
+    password: Optional[str] = Field(default=None, exclude=True)
```

Comment:
Minor: Should it be [secretstr](https://docs.pydantic.dev/dev/api/types/#pydantic.types.SecretStr)?